### PR TITLE
Less randomness and race conditions in tests

### PR DESCRIPTION
**Pull request by privat**
Parallel executions can cause some race collisions on named resources (e.g. DB table names).

To solve this issue, `tests.sh` and `nitunit` initialize  `NIT_TESTING_ID` with a distinct integer identifier that can be used to give unique names to resources. 

`rand` is not a recommended way to get a distinct identifier because its randomness is disabled by default.

### CODE REVIEW

**Comment by xymus** [comment]
+1

File: tests/test_nitcorn.nit [hunk 1]
```diff
@@ -17,8 +17,8 @@ import pthreads
 
 redef class Sys
 	var iface: String is lazy do
-		srand
-		return "localhost:{10000+20000.rand}"
+		var testid = "NIT_TESTING_ID".environ.to_i
+		return "localhost:{10000+testid}"
```
Comment (jcbrinfo) [inline tests/test_nitcorn.nit]: Many edge cases here:
* You may get a port outside the [ephemeral port range](https://en.wikipedia.org/wiki/Ephemeral_port).
* You may collide with a non-Nitcorn process.

You should let the OS chose a port for you.

**Comment by jcbrinfo** [comment]
:+1: +1

File: tests/test_nitcorn.nit (hunk 1 above)
Comment (privat) [inline tests/test_nitcorn.nit]: Yeah, we should let the OS assign a port (cf #2087); we should also libify the nitcorn testing bits (cf #2098).
But I was tired of the recent race conditions so I just take the quick and dirty path.
